src/body/body.rs

```diff
@@ -93,6 +94,28 @@ impl Body {
             tx: tx,
         };
         let rx = Body::new(Kind::Chan {
+            content_length: None,
+            abort_rx: abort_rx,
+            rx: rx,
+        });
+
+        (tx, rx)
+    }
+
+    /// Create a `Body` stream on indeterminate length with an associated sender half.
+    ///
+    /// Useful when wanting to stream chunks from another thread.
+    #[inline]
+    pub fn channel_with_length(content_length: u64) -> (Sender, Body) {
```

Comment:
For now, I'd suggest making this just `pub(crate)`.

Oh and then, these 2 functions can probably be reduced to calling a 3rd, private constructor, `fn new_channel(content_length: Option<u64>)`, to remove duplicate code, yea?